libraries/ESP8266WiFi/examples/BearSSL_MaxFragmentLength/BearSSL_MaxFragmentLength.ino

```diff
@@ -17,17 +18,22 @@ const char *pass = STAPSK;
 void fetch(BearSSL::WiFiClientSecure *client) {
   client->write("GET / HTTP/1.0\r\nHost: tls.mbed.org\r\nUser-Agent: ESP8266\r\n\r\n");
   client->flush();
-  uint32_t to = millis() + 5000;
+  using oneShot = esp8266::polledTimeout::oneShot;
+  oneShot timeout(5000);
   do {
     char tmp[32];
-    memset(tmp, 0, 32);
     int rlen = client->read((uint8_t*)tmp, sizeof(tmp) - 1);
     yield();
     if (rlen < 0) {
       break;
     }
+    if (rlen == 0) {
+      delay(10); // Give background processes some time
+      continue;
```

Comment:
is `delay()` required ?
There is no background process except from receiving, storing, but without processing data.
Calling `read()` often ensures processing them as soon as data are received.

Reply:
If there is no data available yet (rlen == 0), without delay I experienced timeout errors. My assumption is that this is because with no data available the loop that keeps the ESP busy reduces almost to:

`
do {
} while (!timeout)
`

Reply:
I did experienced timeout error @80MHz even with `delay(10)` or with master. @earlephilhower reminded me that SSL works better @160MHz because "mbed uses EC which is insanely slow".
I had no timeout without `delay(10)` (your PR) @160MHz.

With delay, average duration over about 20 request is 273.5ms, and 249.6ms without (measuring the while loop).

Can you retry at 160MHz and see if you still have those timeouts (that I have with master @80Mhz) ?

To be honest, I am concerned about this delay and the associated comment which is wrong to me.

Reply:
> My assumption is that this is because with no data available the loop that keeps the ESP busy reduces almost to: `do { } while (!timeout)`

I agree with saying the arduino infinite loop way of coding is wrong. In that case we could "delay-and-reduce-cpu-activity-until-a byte-is-received" and this api/call is lacking in our API. We are not in an RTOS, but maybe something is doable for this common case.

Reply:
@d-a-v  

> Can you retry at 160MHz and see if you still have those timeouts (that I have with master @80Mhz) ?

I always run the ESP8266 @ 160MHz when using BearSSL, so I experienced the timeout without the delay @ 160MHz and never tried @ 80MHz.

> To be honest, I am concerned about this delay and the associated comment which is wrong to me.

I asume you are not looking for just changing the text of the comment?

Reply:
I too can be picky :) There are already two approvals and changes worksforme (and it's nice). So I let it go as-is. When my pickyness will be boiling I'll make a proposal for the comment-only.

Reply:
;-)